src/ns.c

```diff
@@ -333,7 +333,9 @@ hsk_ns_onrecv(
   // The synth name then resolves to an A/AAAA record that is derived
   // by decoding the name itself (it does not have to be looked up).
   bool should_cache = true;
-  if (strcmp(req->tld, "_synth") == 0 && req->labels <= 2) {
+  if (memcmp(req->tld, "\x06""_synth", 7) == 0
```

Comment:
I don't think the last 0x00 matters really because we have the length byte already and `req->tld` wouldn't ever end with anything else. But, sure! added in f5200ab1251950b897e8ef3f71f5a11737aeaa5a